### PR TITLE
chore: Updating frontend dependencies

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
Updates ReactMarkdown
Updates TypeScript
Updates moment-timezone
Cleans up some ESLint and TypeScript issues
_Include a summary of changes then remove this line_

### Tests
Updated tests
_What tests did you add or modify and why? If no tests were added or modified, explain why. Remove this line_

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/interfaces/Dashboard.ts [hunk 1]
```diff
@@ -14,8 +14,8 @@ export interface DashboardMetadata {
   group_name: string;
   group_url: string;
   last_run_state: string;
-  last_run_timestamp: number;
-  last_successful_run_timestamp: number;
+  last_run_timestamp: number | null;
```
Comment (allisonsuarez) [inline amundsen_application/static/js/interfaces/Dashboard.ts:17]: Are there actually instances where this data would be missing? if so why?

Comment (Golodhros) [reply amundsen_application/static/js/interfaces/Dashboard.ts:17]: It seems so, we have been checking for it all over the place.
Why, no idea!

Comment (allisonsuarez) [reply amundsen_application/static/js/interfaces/Dashboard.ts:17]: that's weird... I would think that we would always get whatever the last good run was

**Review by dikshathakur3119 — APPROVED** [review]

File: amundsen_application/static/js/interfaces/Dashboard.ts (hunk 1 above)
Comment (Golodhros) [reply amundsen_application/static/js/interfaces/Dashboard.ts:17]: Maybe that's the case now, but I haven't dig deeper, I just want to get rid of that typing issue.

**Pull request merged**